corehq/apps/campaign/static/campaign/js/async_configurable_report.js

```diff
@@ -0,0 +1,428 @@
+hqDefine("campaign/js/async_configurable_report", [
+    'jquery',
+    'underscore',
+    'bootstrap5',
+    'hqwebapp/js/bootstrap5/alert_user',
+    'reports/js/charts/main',
+    'reports/js/filters/bootstrap5/main',
+    'reports/js/util',
+    'hqwebapp/js/bootstrap5/main',
+    'hqwebapp/js/initial_page_data',
+    'reports/js/bootstrap5/hq_report',
+    'reports/js/bootstrap5/report_config_models',
+    'reports/js/bootstrap5/standard_hq_report',
+    // 'userreports/js/base',
+    'commcarehq',
+], function (
+    $,
+    _,
+    bootstrap,
+    alertUser,
+    chartsMain,
+    filtersMain,
+    reportsUtil,
+    hqMain,
+    initialPageData,
+    hqReport,
+    reportConfigModels,
+    standardHQReportModule,
+) {
+    return function (o) {
+        var self = {};
+        self.widgetFilters = o.widgetFilters;
+        self.reportContent = o.reportContent;
+        self.filterForm = o.filterForm;
+        // self.loadingIssueModalElem = self.reportContent.find('#loadingReportIssueModal-' + o.reportId);
+        // self.loadingIssueModal = new bootstrap.Modal(self.loadingIssueModalElem.get(0));
+        self.issueAttempts = 0;
+        self.urlRoot = o.urlRoot;
+        
+        // Make sure urlRoot ends with a slash
+        if (self.urlRoot && !self.urlRoot.endsWith('/')) {
```

Comment:
This logic seems to show up in a bunch of places, do you know in what cases the URL will or won't have a slash?